scipy/signal/windows/windows.py

```diff
@@ -1640,6 +1641,60 @@ def exponential(M, center=None, tau=1., sym=True):
     return _truncate(w, needs_trunc)
 
 
+def taylor(N, nbar=4, sll=-30):
+    """
+    Return the Taylor window.
+
+    The Taylor window allows for a selectable sidelobe suppression with a 
+    minimum broadening. This window is commonly used in synthetic aperture radar
+    processing.
+
+    Parameters
+    ----------
+    M : int
+        Number of points in the output window. If zero or less, an
+        empty array is returned.
+    nbar : int
+        Number of nearly constant level sidelobes adjacent to the mainlobe
+    sll : float
+        Desired peak sidelobe level in decibels (db) relative to the mainlobe
+
+    Returns
+    -------
+    out : array
+        The window, with the maximum value normalized to one (the value
+        one appears only if the number of samples is odd).
+
+    See Also
+    --------
+    kaiser, bartlett, blackman, hamming, hanning
+
+    Notes
+    -----
+    The Kaiser window is described by Carrara, Goodman, and Majewski 
+    in 'Spotlight Synthetic Aperture Radar: Signal Processing Algorithms'
+    Page 512-513
+    """
+    B = 10**(-sll/20)
+    A = np.log(B + np.sqrt(B**2 - 1))/np.pi
+    s2 = nbar**2 / (A**2 + (nbar - 0.5)**2)
+    ma = np.arange(1,nbar)
+    def calc_Fm(m):
+        numer = (-1)**(m+1) * np.prod(1-m**2/s2/(A**2 + (ma - 0.5)**2))
+        denom = 2* np.prod([1-m**2/j**2 for j in ma if j != m])
+        return numer/denom
+    
+    Fm = np.array([calc_Fm(m) for m in ma])
+    def W(n):
+        return 2*np.sum(Fm * np.cos(2*np.pi*ma*(n-N/2 + 1/2)/N)) + 1
+    
+    w = np.array([W(n) for n in range(N)])
```

Comment:
this should be vectorized

Reply:
Do you mean using np.vectorize? I've added that now.

Reply:
No I mean you should be able to do `w = W(np.arange(N))`. If it doesn't work, refactor `W` so it can operate on an array.